src/ui.rs

```diff
@@ -26,13 +26,33 @@ fn calculate_geometry(window: &Node, output: &Node, args: Arc<Args>) -> (i32, i3
     (rel_x - anchor_x, rel_y - anchor_y)
 }
 
-fn handle_keypress(conn: Arc<Mutex<Connection>>, key_to_con_id: &HashMap<char, i64>, keyval: &str) {
+fn handle_keypress(
+    conn: Arc<Mutex<Connection>>,
+    key_to_con_id: &HashMap<char, i64>,
+    keyval: &str,
+    command: &Commands,
+) {
     if keyval.len() == 1 {
         // we can unwrap because the keyval has one character
         let c = keyval.chars().next().unwrap();
         if c.is_alphabetic() && c.is_lowercase() {
             let con_id = key_to_con_id[&c];
-            sway::focus(conn, con_id);
+
+            match &command {
+                Commands::Focus => {
+                    sway::focus(&conn, con_id);
```

Comment:
Is there a particular reason why you made `focus` and `swap` take non-owning references to `conn` instead of just cloning? Since it's an `Arc`, the clone should be pretty cheap and would also help us avoid lifetime issues later on (maybe?).

As a note, it's been a while since I've last thought about Rust so I might be missing something.

Reply:
I reverted my changes and used `clone` instead.